rules/sort-interfaces.ts

```diff
@@ -178,6 +202,13 @@ export default createEslintRule<Options<string[]>, MESSAGE_ID>({
                 return accumulator
               }
 
+              if (partitionComment) {
```

Comment:
[nit] Really small nitpicking but I would put this check along with the one related to `options.partitionByNewLine` line 249 as those two concepts are related.